egs/hkust/s5b/local/hkust_data_prep.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+
+
+. path.sh
+
+if [ $# != 2 ]; then
+   echo "Usage: hkust_data_prep.sh AUDIO_PATH TEXT_PATH"
+   exit 1;
+fi
+
+HKUST_AUDIO_DIR=$1
+HKUST_TEXT_DIR=$2
+
+train_dir=data/local/train
+dev_dir=data/local/dev
+
+
+case 0 in    #goto here
+    1)
+;;           #here:
+esac
+
+mkdir -p $train_dir
+mkdir -p $dev_dir
+
+#data directory check
+if [ ! -d $HKUST_AUDIO_DIR ] || [ ! -d $HKUST_TEXT_DIR ]; then
+  echo "Error: run.sh requires two directory arguments"
+  exit 1;
+fi
+
+#find sph audio file for train dev resp.
+find $HKUST_AUDIO_DIR -iname "*.sph" | grep -i "audio/train" > $train_dir/sph.flist
+find $HKUST_AUDIO_DIR -iname "*.sph" | grep -i "audio/dev" > $dev_dir/sph.flist
+
+n=`cat $train_dir/sph.flist $dev_dir/sph.flist | wc -l`
+[ $n -ne 897 ] && \
+  echo Warning: expected 897 data data files, found $n
+
+
+#Transcriptions preparation
+
+#collect all trans, convert encodings to utf-8,
+find $HKUST_TEXT_DIR -iname "*.txt" | grep -i "trans/train" | xargs cat |\
+  iconv -f GBK -t utf-8 - | perl -e '
+    while (<STDIN>) {
+      @A = split(" ", $_);
+      if (@A <= 1) { next; }
+      if ($A[0] eq "#") { $utt_id = $A[1]; } 
+      if (@A >= 3) {
+        $A[2] =~ s:^([AB])\:$:$1:; 
+        printf "%s-%s-%06.0f-%06.0f", $utt_id, $A[2], 100*$A[0] + 0.5, 100*$A[1] + 0.5; 
+        for($n = 3; $n < @A; $n++) { print " $A[$n]" }; 
+        print "\n"; 
+      }
+    }
+  ' | sort -k1 > $train_dir/transcripts.txt 
+
+find $HKUST_TEXT_DIR -iname "*.txt" | grep -i "trans/dev" | xargs cat |\
+  iconv -f GBK -t utf-8 - | perl -e '
+    while (<STDIN>) {
+      @A = split(" ", $_);
+      if (@A <= 1) { next; }
+      if ($A[0] eq "#") { $utt_id = $A[1]; } 
+      if (@A >= 3) {
+        $A[2] =~ s:^([AB])\:$:$1:; 
+        printf "%s-%s-%06.0f-%06.0f", $utt_id, $A[2], 100*$A[0] + 0.5, 100*$A[1] + 0.5; 
+        for($n = 3; $n < @A; $n++) { print " $A[$n]" }; 
+        print "\n"; 
+      }
+    }
+  ' | sort -k1  > $dev_dir/transcripts.txt
+
+
+
+#transcripts normalization and segmentation 
+#(this needs external tools),
+#Download and configure segment tools   
+pyver=`python --version 2>&1 | sed -e 's:.*\([2-3]\.[0-9]\+\).*:\1:g'`
+export PYTHONPATH=$PYTHONPATH:`pwd`/tools/mmseg-1.3.0/lib/python${pyver}/site-packages
+if [ ! -d tools/mmseg-1.3.0/lib/python${pyver}/site-packages ]; then
+  echo "--- Downloading mmseg-1.3.0 ..."
+  echo "NOTE: it assumes that you have Python, Setuptools installed on your system!"
+  wget -P tools http://pypi.python.org/packages/source/m/mmseg/mmseg-1.3.0.tar.gz 
+  tar xf tools/mmseg-1.3.0.tar.gz -C tools
+  cd tools/mmseg-1.3.0
+  mkdir -p lib/python${pyver}/site-packages
+  python setup.py build 
+  python setup.py install --prefix=.
+  cd ../..
+  if [ ! -d tools/mmseg-1.3.0/lib/python${pyver}/site-packages ]; then
```

Comment:
In general I don't like automatically installing stuff without requiring user interaction, because it's a security risk.   What I would be happier with is adding a script to install this like tools/extras/install_mmseg.sh, and (maybe near the top of run.sh as well as inside this script) prompting the user to invoke that if it's not installed.  But if you really don't have time for this, I'd consider just adding  warning in a comment near the top of run.sh that mentions that this is automatically installed.